src/components/HomePage/Forms/MovieForm.tsx

```diff
@@ -1,76 +1,119 @@
-                      onChangeCallBack={setPosterPath}/>
-      <SelectInputGenre defaultSelected={genres} onChangeCallBack={setGenres}/>
-      <InputWithLabel id="movieOverview" label="Overview" name="text" type="textarea" placeholder="Overview here" size="sm" defaultValue={overview}
-                      onChangeCallBack={setOverview}/>
-      <InputWithLabel id="movieRuntime" label="Runtime" type="number" placeholder="Runtime here" size="sm" defaultValue={runtime.toString()}
-                      onChangeCallBack={setRuntime}/>
-      <div className="d-flex justify-content-md-end">
-        <ResetButton />
-        <SubmitButton buttonText={existedId ? 'Save' : 'Submit'} />
-      </div>
-    </Form>
-  );
+    return (
+        <Formik initialValues={{
+            title: movieInfo ? movieInfo.title : "",
+            releaseDate: movieInfo ? movieInfo.releaseDate : "",
+            posterPath: movieInfo ? movieInfo.posterPath : "",
+            overview: movieInfo ? movieInfo.overview : "",
+            runtime: movieInfo ? movieInfo.runtime : "",
+            genres: movieInfo ? movieInfo.genres : []
+        }}
+            validationSchema={formSchema}
+            onSubmit={(values) => {
+                if (movieInfo?.id) {
+                    dispatch(updateMovieCreator({
+                        id: movieInfo.id,
+                        title: values.title,
+                        releaseDate: values.releaseDate,
+                        posterPath: values.posterPath,
+                        runtime: +values.runtime,
+                        overview: values.overview,
+                        genres: values.genres
+                    }));
+                } else {
+                    dispatch(addMovieCreator({
+                        title: values.title,
+                        releaseDate: values.releaseDate,
+                        posterPath: values.posterPath,
+                        runtime: +values.runtime,
+                        overview: values.overview,
+                        genres: values.genres
+                    }));
+                    dispatch(toggleCurrentGenresCreator());
+                }
+                if (onClickCallBack) {
+                    onClickCallBack();
+                }
+            }}
+            onReset={(_, formikHelpers) => {
+                formikHelpers.setFieldValue("title", movieInfo ? movieInfo.title : "");
+                formikHelpers.setFieldValue("releaseDate", movieInfo ? movieInfo.releaseDate : "");
+                formikHelpers.setFieldValue("posterPath", movieInfo ? movieInfo.posterPath : "");
+                formikHelpers.setFieldValue("runtime", movieInfo ? movieInfo.runtime : "");
+                formikHelpers.setFieldValue("overview", movieInfo ? movieInfo.overview : "");
+                formikHelpers.setFieldValue("genres", movieInfo ? movieInfo.genres : []);
```

Comment:
Could we set all values at once?

Reply:
Fixed in commit 144e7682797a8b327df9c556453f21aad1d10993